### DISCUSSION TITLE
异步api发不出消息

### DISCUSSION THREAD

**Discussion by jiangdengke** [simple-robot/Discussions#160] [Q&A]
### 涉及的编程语言

Java

### 涉及的组件库

mirai (QQ)

### 疑问描述

在今天早上出现了异步api发不出消息的情况。发现用的阻塞api还可以发，于是将异步api'改为阻塞api后，相关功能就能用了

**Comment by ForteScarlet** [comment]
使用异步API时，使用 `CompletableFuture.exceptionly` 或 `CompletableFuture.whenComplete` 等可以处理异步结果或异步异常的函数来验证异步任务是否真的执行成功，以及对过程中可能产生的异常进行记录或处理。
